src/main/java/org/truffleruby/core/symbol/SymbolTable.java

```diff
@@ -45,13 +44,13 @@
 
     private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-    // Cache searches based on String
-    private final Map<StringKey, Reference<DynamicObject>> stringSymbolMap = new WeakHashMap<>();
-    // Cache searches based on Rope
-    private final Map<RopeKey, Reference<DynamicObject>> ropeSymbolMap = new WeakHashMap<>();
-    // Weak set of Symbols, SymbolEquality implements equality based on inner rope, to be able to
-    // deduplicate symbols
-    private final Map<SymbolEquality, Reference<DynamicObject>> symbolSet = new WeakHashMap<>();
+    // A cache for j.l.String to Symbols. Entries might get GC'd quickly as nothing references the
+    // StringKey. However, this doesn't matter as the cache entries will be re-created when used.
+    private final Map<StringKey, WeakReference<DynamicObject>> stringSymbolMap = new WeakHashMap<>();
```

Comment:
It's unclear to me if we should keep this.
At least WeakReference seems wrong, as at every GC this table will be emptied.
Maybe SoftReference would make sense?

If `getSymbol(String)` was only called with literal j.l.String we could avoid the StringKey and then not even need this to be weak. But currently this is not the case and `getSymbol(String)` is used with method names and in many other places.

Reply:
Could be a `SoftReference` yes. `WeakReference` isn't 'wrong', as in 'incorrect'.

Reply:
Right, it seems sub-optimal rather.